src/realm/transaction.cpp

```diff
@@ -570,6 +570,7 @@ void Transaction::upgrade_file_format(int target_file_format_version)
     if (current_file_format_version < 24) {
         for (auto k : table_keys) {
             auto t = get_table(k);
+            t->free_collision_table();
```

Comment:
what was the consequence of not doing this?

Reply:
Not sure what the question is. But the collision table is not needed anymore. We will not generate the unresolved key based on the primary key.

Reply:
I see. I thought it was a fix.